backend/de.metas.report/metasfresh-report-service/src/main/java/de/metas/report/jasper/OrgLogoLocalFileLoader.java

```diff
@@ -110,30 +116,81 @@ private File createAndGetLocalLogoFile(@NonNull final OrgId adOrgId)
 		return logoFile;
 	}
 
-	private I_AD_Image retrieveLogoImage(@NonNull final OrgId adOrgId)
+	private I_AD_Image retrieveImage(@NonNull final ResourceNameContext rsc)
 	{
+		final OrgId adOrgId = rsc.getOrgId();
 		if (adOrgId.isAny())
 		{
 			return null;
 		}
 
-		final Properties ctx = Env.getCtx();
+		if (rsc.isLogo())
+		{
+			return retrieveLogoImage(adOrgId);
+		}
+		else
+		{
+			return retrieveOtherImage(rsc);
+		}
+	}
+
+	private I_AD_Image retrieveOtherImage(@NonNull final ResourceNameContext rsc)
+	{
+		final OrgId adOrgId = rsc.getOrgId();
+		final String resourceName = rsc.getResourceName();
+
+		final List<String> listResourceName = Splitter.on('/')
+				.trimResults()
+				.omitEmptyStrings()
+				.splitToList(resourceName);
+
+		if (listResourceName.size() > 0)
+		{
+			final String resource = listResourceName.get(listResourceName.size() - 1);
+			final int indexOf = resource.indexOf(".");
+			if (indexOf < 0)
+			{
+				return null;
+			}
 
+			final String columnName = resource.substring(0, indexOf);
+
+			//
+			// Get image
+			final I_AD_OrgInfo orgInfo = orgsRepo.retrieveOrgInfoRecordOrNull(adOrgId, ITrx.TRXNAME_None);
```

Comment:
you are using the method `retrieveOrgInfoRecordOrNull` ..but AFAIU InterfaceWrapperHelper will throw an NPE if the model is null